src/pages/templates/pages/reader.html

```diff
@@ -201,6 +201,17 @@ <h2 class="modal-title">Image Details</h2>
             },
 
             api: {
+                resourceReady: function() {
+                    console.debug("D2Reader finished loading the resource at: " + new Date());
+                    $(document).ready(function () {
+                        buildTableOfContents();
+                        buildAnnotationList();   
+                        // We only need to create the prefs editor once for a page
+                        if(typeof clusivePrefs !== undefined) {
+                            clusivePrefs = createClusivePrefsEditor();                        
+                        }                                                   
```

Comment:
This feels backwards - don't you want to create it if the variable IS undefined, and leave it alone if it's already set?

Also, feels like building the TOC and annotation list also only need to be done once.

Reply:
Yes, test was backwards; now fixed.

I've created a `firstResourceLoaded` boolean flag we can use to do things only once on the page when a resource is first completely loaded by the Reader, rather than every time a new resource is loaded within the document.